### ISSUE TITLE
Image annotation (areas, rectangles)

### ISSUE THREAD

**Issue by klemay** [hypothesis/product-backlog#669]
## Feature Request Form
Feature request from Zendesk: https://hypothesis.zendesk.com/agent/tickets/2563

Image annotation is mentioned in https://github.com/hypothesis/vision/issues/51, but not anywhere in our backlog that I can find. 

### Problem you are trying to address with this feature
Currently annotations can only anchor to lines of text or to pages (as Page Notes). There is no way to annotate images or areas of pages. 

### Your solution
User has requested the ability to highlight an area of a page / document by creating a rectangle or box. There are other ways to implement, but this is how this user would expect the feature to work.

**Comment by nichtich** [comment]
Some commercial web annotation tools like [zipboard](https://zipboard.co/) support this but require a browser plugin to create a screenshot to draw on. Another option would be to annotate HTML elements such as it was implemented in [Juntoo](https://juntoo.co/).

**Comment by klemay** [comment]
Another user request: https://hypothesis.zendesk.com/agent/tickets/3198

> do you have any plans to allow for other types of notation, such as drawing on a document, or circling something in an image?

**Comment by mkdir-washington-edu** [comment]
https://hypothesis.zendesk.com/agent/tickets/6022
This user is an art historian!

**Comment by mkdir-washington-edu** [comment]
https://hypothesis.zendesk.com/agent/tickets/6595
Image annotation specific to PDFs

https://hypothesis.zendesk.com/agent/tickets/7272

**Comment by indigetal** [comment]
Just put a request in for this myself (Ticket #7272) and was alerted to this existing case, but it looks like it's been a known issue since at least 2014. There is no way to draw a marquee selection in hypothes.is to annotate images or un-selectable text, such as in some PDF files.

**Comment by nichtich** [comment]
The underlying technology, by the way, is established as [IIIF Image API](https://iiif.io/api/image/).

**Comment by mkdir-washington-edu** [comment]
We had a request today on Jeremy/Nate's LMS grading demo.

**Comment by mkdir-washington-edu** [comment]
Other requests:
https://hypothesis.zendesk.com/agent/tickets/7387
https://hypothesis.zendesk.com/agent/tickets/9400
https://hypothesis.zendesk.com/agent/tickets/11169
https://hypothesis.zendesk.com/agent/tickets/11770
https://app.hubspot.com/contacts/6291320/ticket/244588999/

**Comment by mkdir-washington-edu** [comment]
https://hypothesis.zendesk.com/agent/tickets/7387
Since PDFs have an image layer and a text layer, it might be worth exploring if we can allow image annotation over a whole PDF.

**Comment by mkdir-washington-edu** [comment]
https://hypothesis.zendesk.com/agent/tickets/7916 
User wants to annotate PDFs without needing to OCR.

**Comment by mkdir-washington-edu** [comment]
https://hypothesis.zendesk.com/agent/tickets/13514

User wants to have students annotate a comic.

**Comment by nicolae-stroncea** [comment]
On my end what i wanted to do was to be able to highlight images so that they would show on the side in the summary and on hypothes.is when I look through my highlights and annotations. I was able to achieve that by just copying the link to the image through a right click and then adding it into the summary through the markdown notation for linking an image. While not an ideal workflow, it helped for my use case. Leaving this here in case anyone else had this issue and wasn't sure how to go about it.

**Comment by FredrikStenstrom** [comment]
Hello and thx for making such an awesome tool. Me and my collegues are collaborating a lot with PDF documents containing drawings and images and Hypothesis has everything we need except from the ability to mark on images in PDF´s.
I understand that the annotations in Hypothesis is textbased and that its hard to code something that recognizez images but isnt i possible to work with some kind of annotation or markings that simply uses a percentage of the width and height of a page to find an aproximate X/Y position. Like 20% from right and 73% from top to mark a note on a PDF page.
Maybe its not that easy but would be so amazing :)

**Comment by robertknight** [comment]
Thanks for the feedback @FredrikStenstrom!

> Me and my collegues are collaborating a lot with PDF documents containing drawings and images and Hypothesis has everything we need except from the ability to mark on images in PDF´s.

In the case of PDFs there is a standard coordinate system that we could use to specify what parts of a page an annotation refers to. Something we would have to figure out is what to display in the sidebar in the place where the annotated text is normally displayed as a quote. I guess the simplest option would be to show something like `(WIDTH x HEIGHT image on page 3)`. With some further work we might be able to get a capture of the relevant PDF content in there.

**Comment by FredrikStenstrom** [comment]
Wow Robert that was the quickest reply ever in these kind of forums. Thanks for takin time and response and also explain your ideas around a solution. I thought I just send out a cry for help and that it would be forgotten but Im amazed of the fast response :)
Your suggestion sounds good and the important thing is that you want to mark out clearly and be able to jump to the markup point by clicking on the comment on the right but the precision doesnt need to be 100% spot on.
Just out of curiosity: How big chance is it that you will complete this kind of function and how far of in the future are we talkin in that case :)
I just discovered Hypothesis so I have no clue about the development plans and road map Im just happy I could find something that works so good and i so smart by recognizing the fingerprint of a PDF so that you only share comments and not have to share the underlying file to the cloud while working on confidential PDF-documents :)
Please keep up the good work!

**Comment by klemay** [comment]
@FredrikStenstrom hello from the Hypothesis product manager! In terms of priority and time frames - this is something we very much want to do. Our biggest customer base right now is in education and this is a popular request. It's not currently scheduled as a near-term item because we have some other education-specific workflows to take care of first, and we are a small (but growing!) team.

**Comment by beckygeorge03** [comment]
![Screen Shot 2021-05-04 at 10 56 38 AM (2) (1)](https://user-images.githubusercontent.com/72111083/117350152-11e1bf00-ae61-11eb-8b07-18471a2e9b1c.png)

**Comment by peterdm** [comment]
Hi @klemay.  We'd also like to add support for image annotation.  

And if at all possible, image reference information via the API as well:

- **web content**: image link or minimally raw html in the anchor which captures the image tag.
- **pdf**: minimally coordinates, but if you create a clipped image for the sidebar, a link to that would be amazing!

Do you anticipate movement on this feature request this year?

**Comment by mkdir-washington-edu** [comment]
Hi @peterdm. Katelyn has moved on to a new opportunity.

You can see our public roadmap here: https://github.com/hypothesis/product-backlog/projects/6. We're a small team with big goals and at this time it's difficult to say when we're be able to work on image annotation.

**Comment by ryanwwest** [comment]
The Obsidian-annotator community (which uses Hypothes.is as a backend to annotate PDFs, ePUBs and web content) would also be very interested in this: https://github.com/elias-sundqvist/obsidian-annotator/issues/40. 

Area/rectangle annotation seems preferable over straight image annotation in case you want to annotate the image and caption, a specific part of an image, or some other non-image component. Though I could see this getting complicated in reflowable formats like websites/ePUBs that switch from mobile to desktop and portrait/landscape views.

**Comment by paloha** [comment]
I am also coming from the Obsidian-annotator community as @ryanwwest mentioned above. I am annotating scientific papers (pdfs) and I am missing area annotations greatly. I would like to highlight a part of the pdf and write a note that sums it up, so that the next time I need read the paper, I can recall what it was about faster. The pdfs may be scans (if very old papers), or I might want to highlight a part of a paragraph along with a figure and its caption. Would be great to have that feature. Thanks a lot.

**Comment by pidgeon777** [comment]
Hello, I wanted to kindly ask if there are updates.

**Comment by robertknight** [comment]
No, nothing has changed. We had planned to do some work on this late last year, but other things took priority.

**Comment by pidgeon777** [comment]
> No, nothing has changed. We had planned to do some work on this late last year, but other things took priority.

Thanks for the prompt reply 👍.

**Comment by binabdilwali** [comment]
Any chance this will be implemented soon?

**Comment by mkdir-washington-edu** [comment]
For the folks subscribed to this issue, we've released image annotation in the Hypothesis web app over PDFs (for now, with more document types coming). You can [see our KB here](https://web.hypothes.is/help/image-annotation-in-the-hypothesis-web-app/) and the [web app change log here](https://web.hypothes.is/help/hypothesis-web-app-change-log/).

There are some additional changes planned, but in the meantime I'm happy to get your feedback on our implementation. Please feel free to open a ticket at support@hypothes.is and I'll get that forwarded to me. Thanks!
